test/model/tools/model_test/main.cpp

```diff
@@ -244,6 +249,146 @@ load_workload(const char *file)
     return workload;
 }
 
+/*
+ * reduce_counterexample --
+ *     Try to find a smaller workload that reproduces a failure.
+ */
+static void
+reduce_counterexample(std::shared_ptr<model::kv_workload> workload, const std::string &home,
```

Comment:
It would be nice to get a "big-picture" explanation of this part offline if you have time?